src/utils.js

```diff
@@ -482,16 +482,39 @@ export function insertElement(elm, doc, target, asLastChildChild) {
   } catch (e) {}
 }
 
+/**
+ * Returns a promise that completes when the given element triggers a 'load' or 'error' DOM event, or when
+ * `timeout` milliseconds have elapsed.
+ *
+ * @param {HTMLElement} element
+ * @param {Number} [timeout]
+ * @returns {Promise}
+ */
+export function waitForElementToLoad(element, timeout) {
+  return new Promise((resolve) => {
+    const onLoad = function() {
+      element.removeEventListener('load', onLoad);
+      element.removeEventListener('error', onLoad);
```

Comment:
I'm not sure if it matters, but it's probably safer to call `clearTimeout` in this function, too.

Reply:
done. fwiw, the `Promise` spec should require `resolve` to do nothing when called a second time.